File: 2025/Company/Company.Web/Modules/Company/PhoneNumberType/PhoneNumberTypeForm.cs
using Serenity.ComponentModel;

namespace Company.Company.Forms;

[FormScript("Company.PhoneNumberType")]
[BasedOnRow(typeof(PhoneNumberTypeRow), CheckNames = true)]
public class PhoneNumberTypeForm
{
    public string Name { get; set; }
    public short IsActive { get; set; }
    public int TenantId { get; set; }
}

File: 2025/Company/Company.Web/Modules/ServerTypes/Company/PhoneNumberTypeForm.ts
import { StringEditor, IntegerEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface PhoneNumberTypeForm {
    Name: StringEditor;
    IsActive: IntegerEditor;
    TenantId: IntegerEditor;
}

export class PhoneNumberTypeForm extends PrefixedContext {
    static readonly formKey = 'Company.PhoneNumberType';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PhoneNumberTypeForm.init)  {
            PhoneNumberTypeForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(PhoneNumberTypeForm, [
                'Name', w0,
                'IsActive', w1,
                'TenantId', w1
            ]);
        }
    }
}